src/components/TodoList.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import Todo from './Todo';
+import TodoForm from './TodoForm';
+
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      todos: [],
+      showState: 'all'
+    };
+  }
+
+  updateTodoList = todo => {
+    this.setState(prevState => ({
+      todos: [todo, ...prevState.todos]
+    }));
+  };
+
+  toggleCompleted = id => {
+    this.setState(state => ({
+      todos: state.todos.map(todo => {
+        if (todo.id === id) return { ...todo, completed: !todo.completed };
+        return { ...todo };
+      })
+    }));
+  };
+
+  populateTodo = showState => {
+    if (showState === 'remaining')
+      return this.state.todos.filter(todo => !todo.completed);
+    else if (showState === 'completed')
+      return this.state.todos.filter(todo => todo.completed);
+    else return this.state.todos;
+  };
+
+  toggleShowState = showState => {
+    this.setState({ showState });
+  };
+
+  render() {
+    const renderTodo = this.populateTodo(this.state.showState);
+    return (
+      <>
+        <div>
+          <button
```

Comment:
If you're keeping all the buttons, form and the list under this component, the name `TodoList` becomes a bit misleading.
Try moving everything except the actual list to `App` component you'll learn a thing or two in the process. :slightly_smiling_face: